src/workflow/ArcanistLandWorkflow.php

```diff
@@ -310,6 +319,24 @@ public function run() {
         ->setShouldPreview($this->preview)
         ->setBuildMessageCallback(array($this, 'buildEngineMessage'));
 
+      // initialize the shadow engine and execute it if uberShadowEngine is initialized
+      if ($uberShadowEngine) {
+        $uberShadowEngine
```

Comment:
Try clone `$engine` object here and add additional parameters later. I'm not 100% sure it will work because I haven't looked into engine internals, but worth to try. This would let us avoid code duplication here and in the beginning of method.
```
if ($uberShadowEngineEnabled) {
  $uberShadowEngine = clone $engine;
  $uberShadowEngine
    ->setRevision($this->uberGetRevision())
    ->setShouldShadow(true)
    ->execute();
```